presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -437,6 +438,11 @@ protected Type visitDereferenceExpression(DereferenceExpression node, StackableA
                     return handleResolvedField(node, resolvedField.get(), context);
                 }
                 if (!scope.isColumnReference(qualifiedName)) {
+                    // try to match it to an enum literal (eg Mood.HAPPY)
```

Comment:
You might want to modify the comment on line 433 as well. Maybe just change it to something generic as `// Handle qualified name`. Then add additional comment for `// Check whether the qualifiedName is a column reference` and here you can just say `// Check whether the qualifiedName is an enum literal`

Reply:
Also the error thrown probably need to be changed as well. If you look at `missingAttributeException` it says "column .. cannot be resolved", which is no longer accurate.

Reply:
Hmm, I do want to keep some of that phrasing since most of the time that will be correct (it's more likely that people are misspelling a column name than an enum literal).
What about: `Column or qualified name '%s' cannot be resolved` ?

Reply:
Maybe just say `Cannot resolve '%s'`. When you tell people what cannot be resolved they'd know whether they meant it to be a column or enum. I agree that most of the time it's probably a column, but whenever it's not this would be a really confusing error message. (People might not know what's a qualified name so we might as well skip that confusion.)

Reply:
That sounds reasonable – one thing though: there's tons of tests that need updating if we change that message, as they all expect to see that particular error message. 
I can update those, but I'm worrying that we might break tests for other plugins/clients that are not in this repo if we change the error message. What do you think?

Reply:
After discussion, keeping "Column %s cannot be resolved" only when the qualifiedName has a single part